runners/local/runner.go

```diff
@@ -85,11 +96,64 @@ func NewRunnerLocal(settings RunnerLocalSettings) *RunnerLocal {
 			// the destination is a MongoDB database otherwise the Options check would have failed
 			cosmosSettings.WitnessMongoConnString = settings.DstConnString
 		}
+		// check and adjust for configurated settings
 		if settings.LoadLevel != "" {
 			btc := getBaseThreadCount(settings.LoadLevel)
-			cosmosSettings.InitialSyncNumParallelCopiers = btc
-			cosmosSettings.NumParallelWriters = btc / 2
+			if settings.InitialSyncNumParallelCopiers != 0 { // override the loadlevel settings if user specifies
+				cosmosSettings.InitialSyncNumParallelCopiers = settings.InitialSyncNumParallelCopiers
+			} else {
+				cosmosSettings.InitialSyncNumParallelCopiers = btc
+			}
+			if settings.NumParallelWriters != 0 {
```

Comment:
Do you really need the comparison with 0 here? When "settings.NumParallelWriters" equals to 0, we want to use the connector-default value, so we need to set "" to 0. In all cases "cosmosSettings.NumParallelWriters = settings.NumParallelWriters" should be valid

Reply:
Applies to others as well